apps/web/src/pages/integrations/components/Modal/NovuIntegrationCard.tsx

```diff
@@ -0,0 +1,162 @@
+import { ChannelTypeEnum, IConfigCredentials } from '@novu/shared';
+import styled from '@emotion/styled';
+import { Group, useMantineColorScheme } from '@mantine/core';
+import { colors, shadows } from '../../../../design-system';
+import { CardStatusBar } from '../CardStatusBar';
+import { IIntegratedProvider } from '../../IntegrationsStorePage';
+import { LimitBar } from '../LimitBar';
+import { getGradient } from '../../../../design-system/config/helper';
+
+export function NovuIntegrationCard({
+  provider,
+  onConnectClick = undefined,
+  ribbonText = 'Starting provider',
+  selected,
+}: {
+  provider: IIntegratedProvider;
+  onConnectClick?: (visible: boolean, create: boolean, provider: IIntegratedProvider) => void;
+  ribbonText?: string;
+  selected: boolean;
+}) {
+  const { colorScheme } = useMantineColorScheme();
+  const logoSrc = provider.logoFileName[`${colorScheme}`];
+  const brightCard =
+    provider.active ||
+    provider.credentials.some((cred: IConfigCredentials) => {
+      return !cred.value;
+    });
+
+  return (
+    <StyledCard
+      dark={colorScheme === 'dark'}
+      active={brightCard}
+      selected={selected}
+      data-test-id="nouv-integration-provider-card"
+      onClick={() => {
+        if (onConnectClick) {
+          onConnectClick(true, true, provider);
+        }
+      }}
+      clickable={onConnectClick !== undefined}
+    >
+      <RibbonWrapper>
+        <Ribbon>{ribbonText}</Ribbon>
+      </RibbonWrapper>
+      <StyledGroup position="apart">
+        <CardHeader>
+          <Logo src={logoSrc} alt={provider.displayName} />
+        </CardHeader>
+        <CardFooter>
+          <LimitBar
+            channel={provider.channel}
+            label={`${provider.channel === ChannelTypeEnum.EMAIL ? 'Email' : 'Sms'} credits used`}
+          />
+          <div
+            style={{
+              marginTop: '10px',
+            }}
+          >
+            <CardStatusBar active={provider.active} />
+          </div>
+        </CardFooter>
+      </StyledGroup>
+    </StyledCard>
+  );
+}
+
+const StyledGroup = styled(Group)`
+  height: 100%;
+  justify-content: space-between;
+  flex-direction: column;
+`;
+
+const RibbonWrapper = styled.div`
+  width: 115px;
+  height: 115px;
+  position: absolute;
+  right: 10px;
+  top: 10px;
+  transform: rotate(45deg);
+`;
+
+const Ribbon = styled.div`
+  background: ${colors.success};
+  font-size: 9px;
+  width: 100%;
+  text-align: center;
+  color: ${colors.white};
+  line-height: 20px;
+  font-weight: bold;
+`;
+
+const CardHeader = styled.div`
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+
+  svg {
+    color: ${colors.B40};
+  }
+`;
+
+const Logo = styled.img`
+  max-width: 140px;
+  max-height: 50px;
+`;
+
+const CardFooter = styled.div`
+  width: 100%;
+`;
+
+const StyledCard = styled.div<{ dark: boolean; active: boolean; clickable: boolean; selected: boolean }>`
```

Comment:
I think there is no problem with stretching as long we create good breakpoints: The higher the screen the more items it can fit in a row. Having the empty space on the right doesn't look better IMO. So would just leave it as is and perhaps do a breakpoint for higher width screens to fit more items there